### PR TITLE
Change the type parsed from a string to a keyword

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -355,21 +355,39 @@ intervalField
...
+    : INTEGER_VALUE
     ;
 
 baseType
-    : identifier
+    : BOOLEAN
+    | TINYINT
+    | SMALLINT
+    | INT
+    | INTEGER
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:376]: MySQL allows `int(11)` like syntax, do we need to be compatible with it?

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:376]: Yes, but this type is currently not supported in queries, only in create table. We can put this in the support of create table PR

**Review by liuyehcf — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -1526,75 +1526,52 @@ private TableName qualifiedNameToTableName(QualifiedName qualifiedName) {
...
-                    }
-                } else if (signature.equalsIgnoreCase("DECIMAL")) {
-                    throw new IllegalArgumentException("Unsupported type specification: " + type.getText());
-                }
-
-                throw new IllegalArgumentException("Unsupported type specification: " + type.getText());
-            }
-
-            if (signature.equalsIgnoreCase("BOOLEAN")) {
+            if (type.baseType().BOOLEAN() != null) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1529]: we can use switch here?  it looks so much if-else

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1529]: No, this is not an enumerable type, but whether the corresponding keyword is null

**Review by Youngwb — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**